apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -0,0 +1,186 @@
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.metrics.DoubleSupplier;
+import co.elastic.apm.agent.metrics.Labels;
+import co.elastic.apm.agent.metrics.MetricRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.stagemonitor.configuration.ConfigurationOption;
+
+import javax.annotation.Nullable;
+import javax.management.JMException;
+import javax.management.MBeanServer;
+import javax.management.ObjectInstance;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeData;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class JmxMetricTracker implements LifecycleListener {
+
+    private static final String JMX_PREFIX = "jvm.jmx.";
+    private final Logger logger;
+    private static final MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+    private final JmxConfiguration jmxConfiguration;
+    private final MetricRegistry metricRegistry;
+
+    public JmxMetricTracker(ElasticApmTracer tracer) {
+        this(tracer, LoggerFactory.getLogger(JmxMetricTracker.class));
+    }
+
+    JmxMetricTracker(ElasticApmTracer tracer, Logger logger) {
+        this.logger = logger;
+        jmxConfiguration = tracer.getConfig(JmxConfiguration.class);
+        metricRegistry = tracer.getMetricRegistry();
+    }
+
+    @Override
+    public void start(ElasticApmTracer tracer) {
+        jmxConfiguration.getCaptureJmxMetrics().addChangeListener(new ConfigurationOption.ChangeListener<List<JmxMetric>>() {
+            @Override
+            public void onChange(ConfigurationOption<?> configurationOption, List<JmxMetric> oldValue, List<JmxMetric> newValue) {
+                List<JmxMetricRegistration> oldRegistrations = registerJmxMetrics(oldValue);
+                List<JmxMetricRegistration> newRegistrations = registerJmxMetrics(newValue);
+
+                List<JmxMetricRegistration> addedRegistrations = new ArrayList<>(newRegistrations);
+                addedRegistrations.removeAll(oldRegistrations);
+
+                List<JmxMetricRegistration> deletedRegistrations = new ArrayList<>(oldRegistrations);
+                deletedRegistrations.removeAll(newRegistrations);
+
+                for (JmxMetricRegistration addedRegistration : addedRegistrations) {
+                    addedRegistration.register(server, metricRegistry);
+                }
+
+                for (JmxMetricRegistration deletedRegistration : deletedRegistrations) {
+                    deletedRegistration.unregister(metricRegistry);
+                }
+            }
+        });
+        for (JmxMetricRegistration registration : registerJmxMetrics(jmxConfiguration.getCaptureJmxMetrics().get())) {
+            registration.register(server, metricRegistry);
+        }
+    }
+
+    private List<JmxMetricRegistration> registerJmxMetrics(List<JmxMetric> jmxMetrics) {
```

Comment:
This action is not really registering (ie adding to the registry), so maybe something like:
```suggestion
    private List<JmxMetricRegistration> compileJmxMetricRegistrationList(List<JmxMetric> jmxMetrics) {
```